### PR TITLE
project-initial-setup

### PR DESCRIPTION
**Pull request by jerrycode06**
I have done the initial setup for [Project 1](https://gitlab.com/SUSE-UIUX/eos/-/wikis/GSoC-2021-Sub-org-at-Python.org:-EOS#project-1). This can be used as a template to add more things to the project. 

## Screenshots
Capture 1
Capture 2

### CODE REVIEW

**Comment by jerrycode06** [comment]
@cyntss @sharmaaditya570191 @en3sis Please give feedback on this so that I can improve this more. There is a lot of work that needs to be done on it but this is for initial setup.

**Comment by sharmaadityaHQ** [comment]
Looks good to me. However, we would like to discuss if we want to use `microbundle-crl` (Refer - https://github.com/developit/microbundle) while moving ahead in the project. Babel and Webpack work under the hood in React applications that are set up using the create-react-app command. Any special reason to use `microbundle-crl`?

**Comment by Abhirup-99** [comment]
@jerrycode06 it would be better to use Github Actions instead of Travis CI.
See this https://eos-community.slack.com/archives/CFS0J7XAN/p1615918901003400

**Comment by jerrycode06** [comment]
> 
> 
> Looks good to me. However, we would like to discuss if we want to use `microbundle-crl` (Refer - https://github.com/developit/microbundle) while moving ahead in the project. Babel and Webpack work under the hood in React applications that are set up using the create-react-app command. Any special reason to use `microbundle-crl`?

Yeah, Webpack comes with great features but As we know `webpack` requires you to create a separate config file to support your webpack build efforts. But when the project size increase and you are adding new features into it that can cause your config to bloat into a form that unmanageable.
As you mentioned `microbundle-crl` or very famous `parcel` are zero-config web application bundler and they bundle your project using nothing but the `package.json`. 
If we are focused on zero-config needed bundler then I think `parcel` makes more sense to me. What are your experiences with `microbundle-crl`? @sharmaaditya570191

**Comment by jerrycode06** [comment]
> 
> 
> @jerrycode06 it would be better to use Github Actions instead of Travis CI.
> See this https://eos-community.slack.com/archives/CFS0J7XAN/p1615918901003400

Thanks, man, I'll do the changes.

**Comment by jerrycode06** [comment]
@cyntss or @en3sis, Can you please merge this PR? or I have to do something more in it?

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "parser": "babel-eslint",
```
Comment (en3sis) [inline .eslintrc:2]: For the ESLint, Prettier  we should use the same rules  we use here: https://gitlab.com/SUSE-UIUX/eos-user-story/-/blob/master/.eslintrc.yml for consistency

File: example/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This example was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (en3sis) [inline example/README.md:1]: Do we need the `example` folder?

**Review by en3sis — COMMENTED** [review]
Do we know if we need to style a lot of things? If so, maybe having SCSS would be a good addition =)

File: example/README.md (hunk 2 above)
Comment (en3sis) [reply example/README.md:1]: I'm also seeing  +39,656  changes, are we tracking `node_modules` ?

**Comment by vinayaksh42** [comment]
@jerrycode06 I tested out this pull request for rendering Animated SVG icons, It seems like it is unable to render animated icons to the browser.
I used the following code in example/src/App.js
Code_VDWZu3GwX5

The output on the browser was as following:
chrome_t9WUJQ439M

File: .eslintrc (hunk 1 above)
Comment (jerrycode06) [reply .eslintrc:2]: Okay I'll do the changes

File: example/README.md (hunk 2 above)
Comment (jerrycode06) [reply example/README.md:1]: I think we need `example`, I've seen many libraries out there which uses the `example` directory for reference like - 
[react-fontawesome](https://github.com/FortAwesome/react-fontawesome).

Comment (jerrycode06) [reply example/README.md:1]: I saw +39,656 changes too, We are not tracking `node-modules`, it is because of `package-lock.json` and `yarn-lock`.

**Comment by jerrycode06** [comment]
@vinayaksh42, yep, it is creating problems with animated ones. I am working on it
Thanks for testing.

**Comment by en3sis** [comment]
We're closing this since we're now going for a different structure.

**Pull request closed without merging**